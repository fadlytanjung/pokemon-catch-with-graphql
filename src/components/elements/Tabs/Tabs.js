import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.scoped.css';

export default function Tabs(props) {
  const { className, value, activeTab, setActiveTab } = props;
  const customClass = [styles.root, className].filter(Boolean).join(' ');
  return (
    <nav className={customClass}>
      {value.map((val, idx) => (
        <a
          className={activeTab === idx ? styles.active : ''}
          key={idx}
          onClick={() => setActiveTab(idx)}>
          {val}
        </a>
      ))}
    </nav>
  );
}

Tabs.defaultProps = {
  activeTab: 0,
  className: '',
  setActiveTab: () => {},
  value: [''],
};

Tabs.propTypes = {
  activeTab: PropTypes.number,
  className: PropTypes.string,
  setActiveTab: PropTypes.func,
  value: PropTypes.arrayOf(PropTypes.string),
};
