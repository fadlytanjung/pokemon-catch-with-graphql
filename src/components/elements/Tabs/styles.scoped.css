.root {
  background-color: #ffffff;
  border-radius: 0.25rem;
  box-shadow: 0 0.25rem 0.5rem rgba(0, 0, 1, 0.075);
  display: flex;
  justify-content: space-around;
  padding: 0 1rem;
}
.root > a {
  cursor: pointer;
  display: block;
  padding: 1rem 1.5rem;
  text-align: center;
}
.root > a.active {
  box-shadow: inset 0 -0.25rem 0 -0.0625rem var(--primary-color);
  font-weight: 800;
}
@media only screen and (max-width: 48em) {
  .root {
    padding: 0;
    width: 100vw;
  }
}
