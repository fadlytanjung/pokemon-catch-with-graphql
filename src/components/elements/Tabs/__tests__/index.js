import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Tabs from '../Tabs';

describe('src/components/elements/Tabs', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Tabs value={['test1', 'test2']} />);
    expect(tree).toMatchSnapshot();
    Tabs.defaultProps.setActiveTab();
  });

  test('handleClick', () => {
    const setActiveTab = jest.fn();
    const result = Tabs({ ...Tabs.defaultProps, setActiveTab });
    result.props.children[0].props.onClick();
    expect(setActiveTab).toHaveBeenNthCalledWith(1, 0);
  });
});
