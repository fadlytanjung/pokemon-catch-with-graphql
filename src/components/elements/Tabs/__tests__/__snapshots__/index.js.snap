// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/Tabs render 1`] = `
<nav
  className=""
>
  <a
    onClick={[Function]}
  >
    test1
  </a>
  <a
    className=""
    onClick={[Function]}
  >
    test2
  </a>
</nav>
`;
